logfile/operations/types/unzip_files.py

```diff
@@ -0,0 +1,137 @@
+"""
+Module contains file operation to unzip files
+"""
+
+import gzip
+import tarfile
+import logging
+from typing import Optional
+from pathlib import Path
+from logfile.operations.operation_base import OperationBase
+
+# pylint: disable=W1203
+# pylint: disable=no-else-return
+
+class UnzipFiles(OperationBase):
+    """
+    This class is responseable for unzipping files
+
+    Instructions:
+        Directory(str):
+            Relative path in work folder or * for current work folder
+        Recursive(str):
+            False takes current folder, true is recursive through current folder and subs
+    """
+    @staticmethod
+    def extract_tar(filepath: str) -> Optional[str]:
+        """
+        Extracting tar file
+
+        Args:
+            filepath(str): Filepath to tar file
+
+        Returns:
+            str: Directory extracted in
+        """
+        try:
+            filepath = Path(filepath)
+
+            extract_to = filepath.parent / filepath.stem
+            extract_to.mkdir()
+
+            filepath = filepath.as_posix()
+            extract_to = extract_to.as_posix()
+            logging.debug(f"Extracting '{filepath}' to '{extract_to}'")
+
+            tar = tarfile.open(filepath)
+            tar.extractall(extract_to)
+            tar.close()
+
+            logging.debug("Deleting file '{filepath}'")
+            Path(filepath).unlink()
+
+            return extract_to
+
+        except OSError as exc:
+            logging.warning(f"Error while extracting file '{filepath}' {exc}")
```

Comment:
What happens when you get an exception?

Reply:
I print a warning.

Reply:
Yeees....., but what happens after that? ;-)

Reply:
If I cant extract it  (the file is corrupted) i cant do anything about it, so i just continue.